### PR TITLE
Validate HCA dataset name at time of creation

### PR DESCRIPTION
**Pull request by quazi-broad**
## Why
We should be creating HCA datasets with valid names, using the regexes documented in the spec.
[Relevant ticket](https://broadinstitute.atlassian.net/browse/DSPDC-1828)

## This PR
Created a function to perform dataset_name validation using the provided regex
Raises a value error with an invalid name
Added a a test to check for failing validation

## Checklist
- [ ] Documentation has been updated as needed.

### CODE REVIEW

File: orchestration/dagster_orchestration/hca_manage/tests/test_dataset_manager.py [hunk 1]
```diff
@@ -52,3 +52,13 @@ def test_create_dataset_with_policy(self):
...
         self.manager.data_repo_client.create_dataset.assert_called_once()
         self.assertEqual(self.manager.data_repo_client.add_dataset_policy_member.call_count, 2)
+
+    def test_validate_dataset_name(self):
+        with self.assertRaises(ValueError):
+            validate_dataset_name(
+                "fake_dataset_name"
+            )
+
+        validate_dataset_name(
```
Comment (aherbst-broad) [inline orchestration/dagster_orchestration/hca_manage/tests/test_dataset_manager.py:62]: ⛏️ These should be separate tests, one for validation for the success path, one for the failing path (i.e, a unit test should focus on testing one thing only)

File: orchestration/dagster_orchestration/hca_manage/dataset.py [hunk 2]
```diff
@@ -62,8 +64,15 @@ def _remove_dataset(args: argparse.Namespace) -> None:
     hca.delete_dataset(dataset_name=args.dataset_name, dataset_id=args.dataset_id)
 
 
+# validate provided name against dataset name regex from the spec
+def validate_dataset_name(dataset_name: str) -> None:
```
Comment (aherbst-broad) [inline orchestration/dagster_orchestration/hca_manage/dataset.py:68]: ⛏️ Prefix this with a `_` to indicate it's an internal method that clients should not be calling.

File: orchestration/dagster_orchestration/hca_manage/dataset.py [hunk 3]
```diff
@@ -62,8 +64,15 @@ def _remove_dataset(args: argparse.Namespace) -> None:
...
+    if not (search(DATASET_NAME_REGEX, dataset_name)):
```
Comment (aherbst-broad) [inline orchestration/dagster_orchestration/hca_manage/dataset.py:69]: ⛏️ These parens might be redundant.
```suggestion
    if not search(DATASET_NAME_REGEX, dataset_name):
```

**Review by aherbst-broad — APPROVED** [review]
lgtm, just need to split that test up and a few other minor cosmetic issues.

**Review by aherbst-broad — APPROVED** [review]
lgtm

**Review by eparkmonster — APPROVED** [review]

File: orchestration/dagster_orchestration/hca_manage/tests/test_dataset_manager.py (hunk 1 above)
Comment (quazi-broad) [reply orchestration/dagster_orchestration/hca_manage/tests/test_dataset_manager.py]: made two separate tests

**Review by raaidbroad — APPROVED** [review]

**Pull request merged**